lambda_functions/automate_s2_nbar_rolling_archive/execute_s2nbar/s3_sync.py

```diff
@@ -0,0 +1,181 @@
+
+    # handling multi vs single granules + variants of each type
+    if not grn_elements:
+        grn_elements = xml_root.findall(search_term[:-1])
+
+    if grn_elements[0].findtext('IMAGE_ID'):
+        search_term = 'IMAGE_ID'
+    else:
+        search_term = 'IMAGE_FILE'
+
+    # required to identify granule metadata in a multigranule archive
+    # in the earlier l1c products
+    processing_baseline = xml_root.findall('./*/Product_Info/PROCESSING_BASELINE')[0].text
+
+    results = {}
+    for granule in grn_elements:
+        gran_id = granule.get('granuleIdentifier')
+        if not pathname.suffix == '.zip':
+            gran_path = str(pathname.parent.joinpath('GRANULE', gran_id, gran_id[:-7].replace('MSI', 'MTD') + '.xml'))
+            root = ElementTree.parse(gran_path).getroot()
+        else:
+            xmlzipfiles = [s for s in archive.namelist() if 'MTD_TL.xml' in s]
+            if not xmlzipfiles:
+                pattern = gran_id.replace('MSI', 'MTD')
+                pattern = pattern.replace('_N' + processing_baseline, '.xml')
+                xmlzipfiles = [s for s in archive.namelist() if pattern in s]
+            mtd_xml = archive.read(xmlzipfiles[0])
+            root = ElementTree.XML(mtd_xml)
+        sensing_time = root.findall('./*/SENSING_TIME')[0].text
+        results[gran_id] = date_parser(sensing_time)
+
+    return results
+
+
+def update_metadatafile(s3_client, yaml_file, bucket_name, key, ymd):
+    with open(yaml_file) as config_file:
+        temp_metadata = yaml.load(config_file, Loader=Loader)
+
+    temp_metadata['image']['bands'].pop('nbart_blue', None)
+    temp_metadata['image']['bands'].pop('nbart_coastal_aerosol', None)
+    temp_metadata['image']['bands'].pop('nbart_contiguity', None)
+    temp_metadata['image']['bands'].pop('nbart_green', None)
+    temp_metadata['image']['bands'].pop('nbart_nir_1', None)
+    temp_metadata['image']['bands'].pop('nbart_nir_2', None)
+    temp_metadata['image']['bands'].pop('nbart_red', None)
+    temp_metadata['image']['bands'].pop('nbart_red_edge_1', None)
+    temp_metadata['image']['bands'].pop('nbart_red_edge_2', None)
+    temp_metadata['image']['bands'].pop('nbart_red_edge_3', None)
+    temp_metadata['image']['bands'].pop('nbart_swir_2', None)
+    temp_metadata['image']['bands'].pop('nbart_swir_3', None)
+    temp_metadata.pop('lineage', None)
+    temp_metadata['creation_dt'] = temp_metadata['extent']['center_dt']
+    temp_metadata['product_type'] = 'S2MSIARD_NBAR'
+    temp_metadata['id'] = str(uuid.uuid4())
+
+    with open(METADATA_FILE, mode='w') as tf:
+        yaml.dump(temp_metadata, tf, default_flow_style=False, Dumper=Dumper)
+
+    # Upload a file
+    response = upload_to_s3(s3_client, METADATA_FILE, bucket_name, key, ymd)
+    if response:
+        LOG.info(f'{METADATA_FILE} file uploaded')
+
+
+def process_main(s3_client, s3_bucket, level1_done_filepath):
+    for done_file in level1_done_filepath:
```

Comment:
I think this is going to iterate over each character in `level1_done_filepath`, not over a list of file paths.

Reply:
This issue is fixed in the latest updates